### DISCUSSION TITLE
Geojson polygon style not working correctly

### DISCUSSION THREAD

**Discussion by mappinggnome** [TerriaJS/terriajs#6317] [General] [answered]
I'm having an issue where when I set the style of a geojson it applies the stroke correctly but not the fill. I've been able to get it to apply the fill correctly if I set the "disableTableStyle" to true, but than I lose the legend info in the workbench. Any thoughts on how to fix this; am I missing some inherent limitation in the code?

Code and image of the map below. 

![image](https://user-images.githubusercontent.com/105613749/168829524-4930eeed-1c8d-4e5f-9ce5-31411363146a.png)

{
      "id": "dfjasklvkclx",
      "type": "geojson",
      "name": "Basalt Rocks",
      "description": "The Dwarves Delved to Greedily",
      "url": "*********", 
      "hideSource": true,
      "disableTableStyle": false,
      "disablePreview": false,
      "disableZoomTo": true,
      "disableOpacityControl": true,
      "disableExport": true,
      "forceCesiumPrimitives": false,
      "style": {
         "fill": "#ff0000",
         "stroke": "#ff0000",
  }

**Comment by nf-s** [comment, marked as answer]
Hi @mappinggnome 

We have just changed how GeoJson styling works. So your example code should work now.

The `style` parameter you are using now only works with Cesium primitives (equivalent to `forceCesiumPrimitives = true`)

If you want to use protomaps/mvt renderer you have to use the `TableStyle` properties. 

For example

```json
{
  "type": "geojson",
  "name": "Test geojson using table style",
  "url": "http://ci.terria.io/main/build/TerriaJS/test/GeoJSON/height.geojson"
  "styles": [
    {
      "id": "User Style",
      "color": { "nullColor": "rgba(255,0,0,1)" },
      "outline": { "null": { "color": "rgba(249,0,0,1)" } },
      "hidden": false
    }
  ],
  "activeStyle": "User Style"
}
```

The `TableStyle` traits are quite messy, but we have a new "Edit Style" mode in Terria - which you can use to generate style JSON

For example:
- Go to http://ci.terria.io/main/
- Load GeoJSON (or CSV) file
- Click "Edit Style"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037496-ee76ca2c-d275-4247-be0d-69c181e71e1f.png">
- Set style to your liking
- Click "show advanced options"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037527-eb76afec-517f-4877-bd5a-dc946f982089.png">
- Click "Copy user stratum to clipboard"
  - <img width="400" alt="image" src="https://user-images.githubusercontent.com/6187649/172037547-5471f26b-2e55-41dd-a9d9-0a3c89835807.png">
- Merge copied JSON into your catalog item JSON

**Note** this feature is brand new and experimental - so you may run into bugs/quirks

**Reply by mappinggnome** [reply]
FANTASTIC! You guys are awesome and thank you for taking the time to respond. I did try the new "Edit Style" mode and it kept throwing up this error. Not sure if it's user error on my part or not, but thought I should let you know regardless. 

![image](https://user-images.githubusercontent.com/105613749/172258370-7ddbb16e-f8c4-4e7d-a3d1-1401f4e01909.png)

Ended up realizing I needed to use the protomaps/mvt renderer since my dataset was so large so got the table styles function to up and running. I'll leave the code I used for posterity.  

            {
          "id":"dsafdsa",
          "type": "geojson",
          "url": "****************",
          "name": "World Geology",
          "description": "The Dwarves Delved to Greedily",
          "hideSource": true,
          "disablePreview": false,
          "disableZoomTo": false,
          "disableOpacityControl": true,
          "opacity":1,
          "hideLegendInWorkbench": false,
          "disableExport": true,
          "disableSplitter": true,
          "disableAboutData":false,
          "forceCesiumPrimitives":false,
          "show":true,
                
          "styles": [{
            "id": "User Style",
            /* there might be a better way of doing this. Here I set the outline and fill color separately */
            "outline":{
                "mapType":"enum",
                "column":"title",
                "enum":[{
                    "value":"BR",
                    "color":"#554359"},
                        {
                    "value":"AR",
                    "color":"#A271E2"},
                        {
                    "value":"GR",
                    "color":"#E29FCF"},
                        {
                    "value":"SS",
                    "color":"#FBE770"},
                        {
                    "value":"LM",
                    "color":"#F4FFA9"},
                        {
                    "value":"MT",
                    "color":"#76AC74"},
                       {
                    "value":"IC",
                    "color":"#FFFFFF"},
                        {
                    "value":"OS",
                    "color":"#5E88B6"},
                        {
                    "value":"CS",
                    "color":"#E5B36D"},          
                    ]},
            "color":{
                "mapType":"enum",
                "colorColumn":"title",
                "nullColor":"#E5B36D",
                "enumColors":[{
                    "value":"BR",
                    "color":"#554359"},
                        {
                    "value":"AR",
                    "color":"#A271E2"},
                        {
                    "value":"GR",
                    "color":"#E29FCF"},
                        {
                    "value":"SS",
                    "color":"#FBE770"},
                        {
                    "value":"LM",
                    "color":"#F4FFA9"},
                        {
                    "value":"MT",
                    "color":"#76AC74"},
                       {
                    "value":"IC",
                    "color":"#FFFFFF"},
                        {
                    "value":"OS",
                    "color":"#5E88B6"},
                        {
                    "value":"CS",
                    "color":"#E5B36D"}],
                    },
            "hidden": false
          }

**Reply by nf-s** [reply]
No problem 🙂 
I fixed the `navigator.clipboard` issue - it should not happen in latest `terriajs` version `8.2.5`
